packages/account/src/Sections/Profile/PhoneVerification/otp-verification.tsx

```diff
@@ -0,0 +1,146 @@
+import { useEffect, useState, useCallback, Fragment } from 'react';
+import PhoneVerificationCard from './phone-verification-card';
+import { Text, InputGroupButton } from '@deriv-com/quill-ui';
+import { Localize, localize } from '@deriv/translations';
+import { observer, useStore } from '@deriv/stores';
+import { useSendOTPVerificationCode, useSettings } from '@deriv/hooks';
+import { convertPhoneTypeDisplay } from 'Helpers/utils';
+import ResendCodeTimer from './resend-code-timer';
+import DidntGetTheCodeModal from './didnt-get-the-code-modal';
+import PhoneNumberVerifiedModal from './phone-number-verified-modal';
+
+type TOTPVerification = {
+    phone_verification_type: string;
+    setOtpVerification: (value: { show_otp_verification: boolean; phone_verification_type: string }) => void;
+};
+
+const OTPVerification = observer(({ phone_verification_type, setOtpVerification }: TOTPVerification) => {
+    const { client, ui } = useStore();
+    const { setVerificationCode } = client;
+    const { data: account_settings, invalidate } = useSettings();
+    const [should_show_phone_number_verified_modal, setShouldShowPhoneNumberVerifiedModal] = useState(false);
+    const [should_show_didnt_get_the_code_modal, setShouldShowDidntGetTheCodeModal] = useState(false);
+    const [otp, setOtp] = useState('');
+    const [is_button_disabled, setIsButtonDisabled] = useState(false);
+
+    const {
+        sendPhoneOTPVerification,
+        phone_otp_error_message,
+        setPhoneOtpErrorMessage,
+        is_phone_number_verified,
+        is_email_verified,
+        sendEmailOTPVerification,
+        requestOnSMS,
+        requestOnWhatsApp,
+        email_otp_error,
+    } = useSendOTPVerificationCode();
+    const { should_show_phone_number_otp } = ui;
+
+    const reInitializeGetSettings = useCallback(() => {
```

Comment:
do we need here useCallback?

Reply:
Yes we need this here if not we're gonna spam reInitializeGetSettings from ResendCodeTimer component 🙏